src/layers/attention.rs

```diff
@@ -72,11 +199,17 @@ impl ScaledDotProductAttention {
         //       layer is working first...
         let model_width = key.dim(3)?;
         let mut attn_scores = query.matmul(&key.transpose(3, 2)?)?;
-        attn_scores = attention_mask.apply_logit_mask(&attn_scores)?;
         let temperature = (model_width as f64).sqrt();
-        let attn_scores = (attn_scores / temperature)?;
+        attn_scores = (attn_scores / temperature)?;
+
+        if let Some(linear_biases) = &self.linear_biases {
+            attn_scores = linear_biases.forward(&attn_scores)?;
+        }
+        attn_scores = attention_mask.apply_logit_mask(&attn_scores)?;
+
```

Comment:
```suggestion

        attn_scores = attention_mask.apply_logit_mask(&attn_scores)?;
```

To make it clearer that applying the logit mask is related to softmax.